cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DistributedStorageProviderNamespaceAdmin.java

```diff
@@ -66,14 +65,16 @@ public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreExcep
     // TODO: CDAP-1519: Create base directory for filesets under namespace home
     // create HBase namespace
     String hbaseNamespace = tableUtil.getHBaseNamespace(namespaceMeta);
-    if (Strings.isNullOrEmpty(namespaceMeta.getConfig().getHbaseNamespace())) {
-      tableUtil.createNamespaceIfNotExists(getAdmin(), hbaseNamespace);
-      return;
-    }
+    try (HBaseAdmin admin = new HBaseAdmin(hConf)) {
```

Comment:
Looking into it.
Caching will be tricky, since caching the same connection for different users will not be doable.